.github/workflows/kubectl-aks.yml

```diff
@@ -258,6 +258,34 @@ jobs:
           export AZURE_NODE_NAME=$(echo '${{ needs.create-aks-cluster.outputs.nodes }}' | jq -r ".[${{ strategy.job-index }}]")
           make integration-test -o kubectl-aks
 
+  documentation-test:
+    name: Test docs
+    runs-on: ubuntu-latest
+    needs: [ build, unit-tests ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '>=1.20.0' # ie requires go 1.20+
+          check-latest: true
+      - name: Get kubectl-aks from artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: kubectl-aks-linux-amd64-tar-gz
+      - name: Prepare kubectl-aks binary
+        shell: bash
+        run: |
+          tar zxvf kubectl-aks-linux-amd64.tar.gz
+          chmod +x kubectl-aks
+          mv kubectl-aks /usr/local/bin
+          ls -la
```

Comment:
I wanted to ensure that the binary that was built in the previous steps of the CI is actually tested since we eventually ship that in the release. But I need to ensure we clean up all other installation for it.

Reply:
Yep but currently there is no way to test the README against the just built binary. See my comment in https://github.com/Azure/kubectl-aks/pull/52#discussion_r1376196826, I think we should separate the README from others docs and the README could only be run for the releases (which makes sense as it describes how to install a released version). WDYT?

Reply:
> currently there is no way to test the README against the just built binary

I realized it is possible to pass env vars to `ie` so it can be used to add custom directory to PATH like I did [here](https://github.com/Azure/kubectl-aks/pull/51/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R110). This takes care of using the built binary by adding path of the current directory where `kubectl-aks` (built by `kubectl-aks` target) is located. In the CI, `kubectl-aks` is download at correct location and used by the tests. Shouldn't that cover all the cases?

Reply:
I don't think it covers also the README case as it doesn't make sense to run `kubectl krew install aks` (which install kubectl-aks in `~/.krew/bin/` and then run `kubectl aks version` using the binary we compiled in the build CI job.

Then, for the commands documentation, we could use such method even if I expect to have just one binary at that time given that we are cleaning up everything we install in the README. So, the only binary available in that host will be the one we downloaded from the artifacts.

Reply:
> I don't think it covers also the README case as it doesn't make sense to run kubectl krew install aks (which install kubectl-aks in ~/.krew/bin/ and then run kubectl aks version using the binary we compiled in the build CI job.

Thanks for catching this. I defined the precedence now [`krew > local > current directory`](https://github.com/Azure/kubectl-aks/pull/51/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R98) and it is working as expected [krew version](https://github.com/Azure/kubectl-aks/actions/runs/6717691754/job/18256101854#step:7:78), [specific version](https://github.com/Azure/kubectl-aks/actions/runs/6717691754/job/18256101854#step:7:108), [installed from source](https://github.com/Azure/kubectl-aks/actions/runs/6717691754/job/18256101854#step:7:159). This also ensures the `kubectl aks --help` is tested against the binary from the PR so any issues will be caught right away.

> So, the only binary available in that host will be the one we downloaded from the artifacts.

Yeah, I think for commands testing, it is much simpler. I feel like with above we should be fine for `README.md` but I am happy to test it separately. What do you say?

Reply:
I still find it too hacky. Now that you added the cleanup step for all the installation methods, I don't see the need of passing the list of paths in a specific order, there will always be only one binary installed at a time.

The only remaining problem is the `kubectl aks help`.... Mmmm I'm not sure what to do. We could either use `./kubectl-aks help` or remove it as the list of commands are already describe above in the documentation.

What do you think?